src/test/hardfork/archive_migration_tests/archive_migration_tests.ml

```diff
@@ -0,0 +1,416 @@
+            conn_str_target_db migration_end_slot
+            ~should_contain_pending_blocks:true
+        in
+        HardForkSteps.compare_replayer_outputs reference_replayer_output
+          actual_replayer_output ~compare_receipt_chain_hashes:false ;
+
+        let%bind blocks =
+          HardForkSteps.download_mainnet_precomputed_blocks steps ~from:11
+            ~num_blocks:10
+        in
+        let%bind _ =
+          HardForkSteps.archive_mainnet_precomputed_blocks steps blocks
+            conn_str_mainnet_source_db
+        in
+        let%bind hash =
+          HardForkSteps.get_max_state_hash
+            (Uri.of_string conn_str_mainnet_source_db)
+        in
+        let%bind migration_end_slot =
+          HardForkSteps.get_migration_end_slot_for_state_hash
+            conn_str_mainnet_source_db (Option.value_exn hash)
+        in
+
+        let%bind _ =
+          HardForkSteps.run_compatible_replayer steps
+            ~archive_uri:conn_str_mainnet_source_db
+            ~input_config:(Filename.basename reference_replayer_input)
+            ~output_ledger:(Filename.basename reference_replayer_output)
+            ~clear_checkpoints:true
+        in
+
+        let%bind _ =
+          HardForkSteps.perform_berkeley_migration steps ~batch_size:2
+            ~genesis_ledger:env.paths.mainnet_genesis_ledger
+            ~source_archive_uri:conn_str_mainnet_source_db
+            ~source_blocks_bucket:env.paths.mainnet_data_bucket
+            ~target_archive_uri:conn_str_target_db
+            ~end_global_slot:(Some migration_end_slot)
+        in
+
+        let%bind _ =
+          HardForkSteps.run_migration_replayer steps
+            ~archive_uri:conn_str_target_db
+            ~input_config:reference_replayer_input ~interval_checkpoint:1
+            ~output_ledger:actual_replayer_output
+        in
+
+        let%bind _ =
+          HardForkSteps.compare_hashes conn_str_mainnet_source_db
+            conn_str_target_db migration_end_slot
+            ~should_contain_pending_blocks:true
+        in
+        HardForkSteps.compare_replayer_outputs reference_replayer_output
+          actual_replayer_output ~compare_receipt_chain_hashes:false ;
+        Deferred.unit )
+end
+
+let env =
+  let doc = "Path Path to env file" in
+  Cmdliner.Arg.(
+    required
+    & opt (some string) None
+    & info [ "env" ] ~doc ~docv:"HARDFORK_TEST_ENV_FILE")
+
+let () =
+  let open Alcotest in
+  run_with_args "Hardfork test suite." env
+    [ ( "mainnet_migration"
+      , [ test_case "Test for short global slots on mainnet data" `Quick
+            HardForkTests.mainnet_migration
+        ] )
+    ; ( "random_migration"
+      , [ test_case "Test for short global slots on artificial data" `Quick
+            HardForkTests.random_migration
+        ] )
+    ; ( "checkpoints"
+      , [ test_case "Test for checkpoint in migration process" `Quick
+            HardForkTests.checkpoint
+        ] )
+    ; ( "incremental"
```

Comment:
For the incremental test, it seems like this has already covered the "checkpoint" test. Can we just have the incremental test?

Reply:
Also for the "incremental" test, we are downloading blocks and then adding them to the genesis database to get the "desired" database for migration. Can we instead precomputed the "desired" database dump, so that running the test would not require us to download precomputed blocks and then add those to the archive db. This would speed up the test and this is closer to what the cron job would actually get.
So I mean instead of get a genesis database and adding precomputed blocks, we would have
1) a database dump with the first 11 blocks
2) a database dump with the first 21 blocks
this should be equivalent to what you did in the "incremental" test

Reply:
To push things one step further, if we make the incremental test to have more blocks, then we could just have this test cover all the above tests.